cmd/olm/manager.go

```diff
@@ -123,6 +123,19 @@ func Manager(ctx context.Context, debug bool) (ctrl.Manager, error) {
 		return nil, err
 	}
 
+	cleanupCsvReconciler, err := operators.NewCleanupCsvReconciler(
```

Comment:
I'll defer to @joelanford and @awgreene on this - while I understand why using the `controller-runtime` approach may be appealing here, it might be premature to move. It's a duplicate in-memory cache of all of these objects, which is a crying shame, and to make the deletion logic robust we'd need to add listers for mutating and validating admission webhooks, which moves us in the wrong direction. Then again, getting this right in the byzantine framework from v0 may be more work than it's worth.

Reply:
This is strongly based on (i.e. copied from) `adoption_controller.go`, which already uses `controller-runtime` and reconciles CSVs already... BUT is only used when a specific feature flag is enabled. So, my original plan to piggy back on an existing CSV reconciler (there are two already!), was abandoned, and I created a new one. The `controller-runtime` seems to be easier to use/manage, especially with finalizers.

Reply:
Yes, I think the ease of coding is certainly on the side of the `controller-runtime` approach - my concern is with duplicating the memory footprint of those informers.

Reply:
Understood; the reconciller in `operatorconditiongenerator_controller.go` already reconciles the CSV, so the impact should be minimal. And I could probably move the code into there.

Reply:
Either one is missing listers for some of these resources and will have the same downside. @joelanford do we care?